### ISSUE TITLE
Training Loop & Loss

### ISSUE THREAD

**Issue by DiaaAhmedHassan** [DiaaAhmedHassan/GPT2Mini#3]
**Responsibilities:**

- [ ] Training loop using torch
- [ ] Optimizer and CrossEntropyLoss
- [ ] Batch processing and padding (if needed)
- [ ] Train on the small novel dataset
